code/game/objects/structures/tank_dispenser.dm

```diff
@@ -1,33 +1,57 @@
+#define MAX_TANK_STORAGE	10
+
 /obj/structure/dispenser
 	name = "tank storage unit"
 	desc = "A simple yet bulky storage device for gas tanks. Has room for up to ten oxygen tanks, and ten plasma tanks."
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "dispenser"
 	density = 1
 	anchored = 1.0
-	var/oxygentanks = 10
-	var/plasmatanks = 10
-	var/list/oxytanks = list()	//sorry for the similar var names
-	var/list/platanks = list()
+	var/starting_oxygen_tanks = MAX_TANK_STORAGE // The starting amount of oxygen tanks the dispenser gets when it's spawned
+	var/starting_plasma_tanks = MAX_TANK_STORAGE // Starting amount of plasma tanks
+	var/list/stored_oxygen_tanks = list() // List of currently stored oxygen tanks
+	var/list/stored_plasma_tanks = list() // And plasma tanks
 
 /obj/structure/dispenser/oxygen
-	plasmatanks = 0
+	starting_plasma_tanks = 0
 
 /obj/structure/dispenser/plasma
-	oxygentanks = 0
+	starting_oxygen_tanks = 0
 
 /obj/structure/dispenser/New()
 	..()
+	initialize_tanks()
 	update_icon()
 
+/obj/structure/dispenser/Destroy()
+	..()
+	QDEL_LIST(stored_plasma_tanks)
+	QDEL_LIST(stored_oxygen_tanks)
+
+/obj/structure/dispenser/proc/initialize_tanks()
+	for(var/I in 1 to starting_plasma_tanks)
+		var/obj/item/tank/plasma/P = new(src)
+		stored_plasma_tanks.Add(P)
+
+	for(var/I in 1 to starting_oxygen_tanks)
+		var/obj/item/tank/oxygen/O = new(src)
+		stored_oxygen_tanks.Add(O)
+
 /obj/structure/dispenser/update_icon()
 	overlays.Cut()
```

Comment:
use the proper procs for overlay adding and cutting please.

`cut_overlays` for instance; should be able to find examples other places.